msg.go

```diff
@@ -364,11 +367,21 @@ Loop:
 						for _, r := range bufs {
 							s = append(s, r)
 						}
+						maxLen += 3 // unprintable char is 3 extra bytes
 					} else {
 						s = append(s, b)
 					}
 				}
 			}
+			// never exceed the allowed label count lenght (63) 
+			if labels >= 63 { 
+				return "", lenmsg, &Error{err: "name exceeds 63 labels"}
+			}
+			labels += 1
+			// never exceed the allowed doman name length (255 octets) 
+			if len(s) >= maxLen {
+				return "", lenmsg, &Error{err: "name exceeded allowed 255 octets"}
```

Comment:
Pointing to the RFC doesn't help substantiate that 256 octet names exist in the wild. I don't think mDNSResponder supports 256 octet names (I haven't been able to get `DNSServiceRegisterRecord` to register such a record) and I believe that covers Android, iOS and macOS. Even if mDNSResponder did, such long names aren't going to be produced unless a developer drops below the high-level DNS-SD APIs offered on those platforms. So to the best of my knowledge 256 octet names exist only in the spec.